app/src/main/java/org/mozilla/fenix/historymetadata/view/RecentlyVisited.kt

```diff
@@ -134,63 +150,98 @@ private fun RecentVisitItem(
         Column(
             modifier = Modifier.fillMaxSize()
         ) {
-            Text(
+            RecentlyVisitedTitle(
                 text = recentVisit.title,
-                modifier = Modifier.padding(top = 7.dp, bottom = 2.dp),
-                color = FirefoxTheme.colors.textPrimary,
-                fontSize = 16.sp,
-                overflow = TextOverflow.Ellipsis,
-                maxLines = 1
+                modifier = Modifier.padding(top = 7.dp, bottom = 2.dp)
             )
 
             RecentlyVisitedCaption(recentVisit.historyMetadata.size)
 
             if (showDividerLine) {
-                Divider(
-                    modifier = Modifier.padding(top = 9.dp),
-                    color = FirefoxTheme.colors.dividerLine,
-                    thickness = 0.5.dp
-                )
+                RecentlyVisitedDivider(modifier = Modifier.padding(top = 9.dp))
             }
         }
 
-        DropdownMenu(
-            expanded = menuExpanded,
-            onDismissRequest = { menuExpanded = false },
-            modifier = Modifier.background(color = FirefoxTheme.colors.surface)
-                .height(52.dp)
-                .scrollable(
-                    state = ScrollState(0),
-                    orientation = Orientation.Vertical,
-                    enabled = false
-                )
-        ) {
-            for (item in menuItems) {
-                DropdownMenuItem(
-                    onClick = {
-                        menuExpanded = false
-                        item.onClick(recentVisit)
-                    },
-                    modifier = Modifier.fillMaxHeight()
-                ) {
-                    Text(
-                        text = item.title,
-                        color = FirefoxTheme.colors.textPrimary,
-                        maxLines = 1,
-                        modifier = Modifier.align(Alignment.Top)
-                            .padding(top = 6.dp)
-                            .scrollable(
-                                state = ScrollState(0),
-                                orientation = Orientation.Vertical,
-                                enabled = false
-                            ).fillMaxHeight()
-                    )
-                }
+        RecentlyVisitedMenu(
+            showMenu = isMenuExpanded,
+            menuItems = menuItems,
+            recentVisit = recentVisit,
+            onLeavingMenu = { isMenuExpanded = false }
```

Comment:
Could we just stick with the DropdownMenu naming for this and rename this to `onDismissRequest`

Reply:
Updated in https://github.com/mozilla-mobile/fenix/pull/22535.